wolfcrypt/src/port/st/stm32.c

```diff
@@ -329,11 +335,11 @@ int wc_Stm32_Hash_Update(STM32_HASH_Context* stmCtx, word32 algo,
     while (len) {
         word32 add;
 
-        /* fill the FIFO plus one additional to flush the block */
-        chunkSz = ((STM32_HASH_FIFO_SIZE + 1) * STM32_HASH_REG_SIZE);
-        /* account for extra bytes in the FIFO (use mask 0x3F to get remain) */
-        chunkSz -= (stmCtx->fifoBytes &
-            ((STM32_HASH_FIFO_SIZE * STM32_HASH_REG_SIZE)-1));
+        chunkSz = blockSize;
```

Comment:
I'll need to re-run some tests on a few STM32 targets to make sure this doesn't break anything. Some of the hashing hardware FIFO 16 * 32-bit words would not save/restore correctly if the FIFO was filled and not flushed. This logic was here to support that edge case. Its possible you've signified the old logic in a good way, but I'll need to confirm before I can merge.

Reply:
Please do, digging into the HAL, I think I see why you did it and I think what I did equates to the same thing. But if it is wrong we can work towards a better way.